pkg/operation/seed/seed.go

```diff
@@ -563,9 +563,9 @@ func RunReconcileSeedFlow(
 			if err != nil {
 				return err
 			}
-			if len(currentResources) != 0 && currentResources[resource] != nil {
+			if len(currentResources) != 0 && currentResources["prometheus"] != nil {
 				monitoringResources[resource] = map[string]interface{}{
-					resource: currentResources[resource],
+					resource: currentResources["prometheus"],
```

Comment:
If I understood the code correctly, there is a mismatch among the three things that should ideally be the same.

1. The values rendering in this function along with [here](https://github.com/gardener/gardener/pull/3998/files#diff-8495b57b25d39744fe1b20b1e1ce695d130f730142af86ff9c5929afcdbabeb4L780-L781) is effectively passing `.Values.aggregatePrometheus.resources.aggregate-prometheus`.
2. The [value.yaml](https://github.com/gardener/gardener/blob/1c3fed84dd2c50aef861b7e0e57f2931c0cbe9fd/charts/seed-bootstrap/values.yaml#L56-L71) of the chart is expecting the structure to be  effectively `.Values.aggregatePrometheus.resources.prometheus`.
3. The [`StatefulSet` YAML](https://github.com/gardener/gardener/blob/1c3fed84dd2c50aef861b7e0e57f2931c0cbe9fd/charts/seed-bootstrap/templates/aggregate-prometheus/statefulset.yaml#L61) is using `.Values.prometheus.resources.prometheus`.